server/app/views/components/FieldWithLabel.java

```diff
@@ -317,7 +317,6 @@ private DivTag getNonNumberInputTag() throws RuntimeException {
 
   /** Public final tag getters * */
   public DivTag getTextareaTag() throws RuntimeException {
-    genRandIdIfEmpty();
```

Comment:
Why remove this call?

Reply:
It'll be called instead in the shared applyAttrsAndGenLabel(...) function below. That shared function (or checkboxApplyAttrsAndGenLabel) will be called by all the public getXTag specific functions.

Reply:
Aha, I'd totally missed that one. Also, the addition of so many tests gave me a lot of confidence that it worked, so thanks for that as well :).